src/Microsoft.AspNetCore.Mvc.Formatters.Xml/ValidationProblemDetailsWrapper.cs

```diff
@@ -0,0 +1,121 @@
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Xml;
+using System.Xml.Serialization;
+
+namespace Microsoft.AspNetCore.Mvc.Formatters.Xml
+{
+    /// <summary>
+    /// Wrapper class for <see cref="ValidationProblemDetails"/> to enable it to be serialized by the xml formatters.
+    /// </summary>
+    [XmlRoot(nameof(ValidationProblemDetails))]
+    public class ValidationProblemDetailsWrapper : ProblemDetailsWrapper, IUnwrappable
+    {
+        private static readonly string ErrorKey = "Mvc-Errors";
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ValidationProblemDetailsWrapper"/>.
+        /// </summary>
+        public ValidationProblemDetailsWrapper()
+            : this(new ValidationProblemDetails())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ValidationProblemDetailsWrapper"/> for the specified
+        /// <paramref name="problemDetails"/>.
+        /// </summary>
+        /// <param name="problemDetails">The <see cref="ProblemDetails"/>.</param>
+        public ValidationProblemDetailsWrapper(ValidationProblemDetails problemDetails)
+            : base(problemDetails)
+        {
+            ProblemDetails = problemDetails;
+        }
+
+        internal new ValidationProblemDetails ProblemDetails { get; }
+
+        /// <inheritdoc />
+        protected override void ReadValue(XmlReader reader, string name)
+        {
+            if (reader == null)
+            {
+                throw new ArgumentNullException(nameof(reader));
+            }
+
+            if (string.Equals(name, ErrorKey, StringComparison.Ordinal))
+            {
+                reader.Read();
+                ReadErrorProperty(reader);
+            }
+            else
+            {
+                base.ReadValue(reader, name);
+            }
+        }
+
+        private void ReadErrorProperty(XmlReader reader)
+        {
+            if (reader.IsEmptyElement)
+            {
+                return;
+            }
+
+            while (reader.NodeType != XmlNodeType.EndElement)
+            {
+                var key = XmlConvert.DecodeName(reader.LocalName);
+                var value = reader.ReadInnerXml();
+                if (string.Equals(EmptyKey, key, StringComparison.Ordinal))
+                {
+                    key = string.Empty;
+                }
+
+                ProblemDetails.Errors.Add(key, new[] { value });
+                reader.MoveToContent();
+            }
+        }
+
+        /// <inheritdoc />
+        public override void WriteXml(XmlWriter writer)
+        {
+            if (writer == null)
+            {
+                throw new ArgumentNullException(nameof(writer));
+            }
+
+            base.WriteXml(writer);
+
+            writer.WriteStartElement(XmlConvert.EncodeLocalName(ErrorKey));
```

Comment:
Suggest returning before this line if `ProblemDetails.Errors.Count == 0`. No need for an empty `<Mvc-Errors />` element.

Reply:
thx